src/utils/mihomo.rs

```diff
@@ -0,0 +1,111 @@
+        let default_payload = "'{\"path\": \"\", \"payload\": \"\"}'".to_string();
+
+    Self {
+        client: reqwest::blocking::Client::new(),
+        api: controller_api,
+        proxy_addr,
+        default_payload,
+    }
+    }
+    fn get(&self, url:&str, payload:Option<&String>) -> Result<String, reqwest::Error>{
+        let api = format!("{}{}", self.api, url);
+        let response = match payload {
+            Some(kv) => self.client.get(format!("{}{}", api, kv)),
+            None => self.client.get(api),
+        }.send();
+        match response {
+            Ok(r) => r.text(),
+            Err(e) => {
+                log::error!("[ClashUtil] exec {} failed! {}", url, e.status().unwrap());
+                Err(e)
+            }
+        }
+    }
+    fn post(&self, url:&str, payload:Option<&String>) -> Result<String, reqwest::Error>{
+        let api = format!("{}{}", self.api, url);
+        let response = match payload {
+            Some(kv) => self.client.post(api).body(kv.to_owned()),
+            None => self.client.post(api),
+        }.send();
+        match response {
+            Ok(r) => r.text(),
+            Err(e) => {
+                log::error!("[ClashUtil] exec {} failed! {}", url, e.status().unwrap());
+                Err(e)
+            }
+        }
+    }
+    fn put(&self, url:&str, payload:Option<&String>) -> Result<String, reqwest::Error>{
+        let api = format!("{}{}", self.api, url);
+        let response = match payload {
+            Some(kv) => self.client.put(api).body(kv.to_owned()),
+            None => self.client.put(api),
+        }.send();
+        match response {
+            Ok(r) => r.text(),
+            Err(e) => {
+                log::error!("[ClashUtil] exec {} failed! {}", url, e.status().unwrap());
+                Err(e)
+            }
+        }
+    }
+
+    pub fn restart(&self) -> Result<String, reqwest::Error>{
+        self.post("/restart", Some(&self.default_payload))
+    }
+    pub fn update_geo(&self) -> Result<String, reqwest::Error>{
+        self.post("/configs/geo", Some(&self.default_payload))
+    }
+    pub fn log(&self) -> Result<String, reqwest::Error>{
+        self.get("/logs", None)
+    }
+    pub fn traffic(&self) -> Result<String, reqwest::Error>{
+        self.get(&"/traffic", None)
+    }
+    pub fn config_reload(&self, payload:String) -> Result<reqwest::blocking::Response, reqwest::Error>{
+        let request = self.client
+          .put(format!("{}/configs?force=true", self.api))
+          .body(payload)
+          .send();
+    request
+    }
+    pub fn config_get(&self) -> Result<String, reqwest::Error>{
+        self.get(&"/configs", None)
+    }
+    pub fn mock_clash_core(&self, url:&str) -> Result<reqwest::blocking::Response, reqwest::Error>{
+        let proxy = reqwest::Proxy::http(&self.proxy_addr).unwrap();
+        let clash_client = reqwest::blocking::Client::builder()
```

Comment:
In ClashUtil, it might be better to add an `Option<reqwest::blocking::Client>` as a cache. Since users' configurations might include multiple proxy and rule file links, some users might open ClashTui just to switch configurations without updating the profile.